partiql-lang/src/test/kotlin/org/partiql/lang/eval/EvaluatingCompilerSelectHeadstrongTest.kt

```diff
@@ -0,0 +1,78 @@
+package org.partiql.lang.eval
+
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ArgumentsSource
+import org.partiql.lang.eval.evaluatortestframework.EvaluatorTestCase
+import org.partiql.lang.util.ArgumentsProviderBase
+
+/** Queries that demonstrate the "headstrong" optimization of SELECT items,
+ *  implemented in [EvaluatingCompiler.compileProjectItemSource]. */
+class EvaluatingCompilerSelectHeadstrongTest : EvaluatorTestBase() {
```

Comment:
I agree the tests do not demonstrate the effect of the optimization, but it happens during their execution.  If the optimization code is merged and runs in production, it's better to have some kind of test for it running in the suite rather than let examples get stale in a doc somewhere.